weavetest/auth.go

```diff
@@ -0,0 +1,76 @@
+package weavetest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/iov-one/weave"
+)
+
+// Auth is a mock implementing x.Authenticator interface.
+//
+// This structure authenticates any of referenced conditions.
+// You can use either Signer or Signers (or both) attributes to reference
+// conditions. This is for the convinience and each time all signers
+// (regardless which attribute) are considered.
+type Auth struct {
+	// Signer represents an authentication of a single signer. This is a
+	// convinience attribute when creating an authentication method for a
+	// single signer.
+	// When authenticating all signers declared on this structure are
+	// considered.
+	Signer weave.Condition
+
+	// Signers represents an authentication of multiple signers.
+	Signers []weave.Condition
+}
+
+func (a *Auth) GetConditions(weave.Context) []weave.Condition {
+	if a.Signer != nil {
+		return append(a.Signers, a.Signer)
+	}
+	return a.Signers
+}
+
+func (a *Auth) HasAddress(ctx weave.Context, addr weave.Address) bool {
+	for _, s := range a.Signers {
+		if addr.Equals(s.Address()) {
+			return true
+		}
+	}
+	return addr.Equals(a.Signer.Address())
```

Comment:
Why separate here? What about range over `a.GetConditions()`

I dislike relying on the handling of `nil.Address`

Reply:
Existence of two signer attributes on this structure - `Signer` and `Signers` is for convenience only. I have created  this structure with only `Signers` at first but then if you want to create an authenticator for only one signer you had to write 
```go
auth := &weavetest.Auth{Signers: []weave.Condition{signer}}
```
In comparison, having a `Signer` attribute you are saving yourself typing of 20 characters!! :joy_cat: 
```go
auth := &weavetest.Auth{Signer: signer}
```
Having two signer references means that you can use both of them at the same time as well.
```go
auth := &weavetest.Auth{
    Signers: signersGroup,
    Signer: signer,
}
```
---

I can change the last line of the test to 
```go
if a.Signer == nil { 
    return false 
} 
return addr.Equals(a.Signer.Address())
```
What do you think?

Reply:
Sure.
I very much like the two members, they make it easier to use. 
I just wanted to avoid the comparison to nil